membrain.cabal

```diff
@@ -50,6 +50,7 @@ common common-options
                        -fhide-source-paths
                        -Wmissing-export-lists
                        -Wpartial-fields
+                       -freduction-depth=250
```

Comment:
Sorry, I can't find documentation for this. (I'm looking into https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/flags.html )
Could you please explain why this is needed?

Reply:
I want to add from my side that adding custom and rare flags (that apparently doesn't have documentation) might be a stopper for merging. This constant might work for natural 7000. But GHC might require a much higher constant for bigger naturals. This can slow down compilation significantly for the `membrain` users or produce difficult errors. The fact that this constant needs to patched to make the library work looks like an undesirable property to me.

Reply:
Hi, answers inline.

> Sorry, I can't find documentation for this

Yes, flag not being documented is in fact a ghc bug: https://gitlab.haskell.org/ghc/ghc/issues/15756

GHC manpage seems to have this documented though: 
https://manpages.debian.org/stretch/ghc/haskell-compiler.1.en.html#Language_options

This flag controls how many steps GHC is allowed to take for simplifying a single type expression.

> Could you please explain why this is needed?

In order to understand whether large Nats are only dividable by 5 or 2, but not by any other number, we need to continuously divide them by 5 and 2 and see if we end up with 1.
This creates a type expression like `(((((7000 Div 5) Div 5) Div  5) Div 5)...` for ghc to simplify and ghc simply has a limit on number of reductions it wants to perform.

For Yobibyte, currently the largest unit, this reduction size is exactly `250`, hence the `-freduction-depth=250` flag.